tests/devices/test_default_qubit_tf.py

```diff
@@ -320,6 +320,33 @@ def test_two_qubit_parameters(self, init_state, op, func, theta, tol):
         expected = func(theta) @ state
         assert np.allclose(res, expected, atol=tol, rtol=0)
 
+    def test_apply_ops_above_8_wires(self, mocker):
+        """Test that when 9 wires are used, the _apply_ops dictionary is empty and application of a
+        CNOT gate is performed using _apply_unitary_einsum"""
+        dev = DefaultQubitTF(wires=9)

```

Comment:
That is an excellent point! We discussed offline and found that the difference is due to you having an earlier version of PyTorch (`v2.0.1`) whereas I was using `v2.3.0`.

I then decided to run the test suite in `v2.0.1` and realized that these special apply operations do not work with this earlier version - there is a failure when calculating the gradient. In fact, the special apply ops are incompatible with versions before 2.3.0 (quite recent!). So I thought the best remedy (done in 60de26d09fdb334db17276cf9e16cb077c5be1b5) is to only use the special apply ops for `v2.3.0` and above.

Reply:
It might be good to consider deprecating TF < 2.1 at some point. I already dislike that our support for TF versions is all over the place depending on the module, and at the same time, v2.0 was notoriously buggy when it comes to complex numbers.